src/tests/ftest/control/dmg_config_generate.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+
+from apricot import TestWithServers
+from command_utils import CommandFailure
+
+
+class ConfigGenerate(TestWithServers):
+    """Test Class Description:
+
+    Verify the veracity of the configuration created by the command and what
+    the user specified, input verification and correct execution of the server
+    once the generated configuration is propagated to the servers.
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a ConfigGenerate object."""
+        super(ConfigGenerate, self).__init__(*args, **kwargs)
+        self._start_servers = False
+
+    def start_server_managers(self):
+        """Start the daos_server processes on each specified list of hosts."""
+        if self._start_servers:
+            super(ConfigGenerate, self).start_server_managers()
+
+    def run_test(self, pmem, nvme, net):
+        """Run test."""
+        self._start_servers = True
+        self.server_managers[-1].manager.job.discover_pmem.value = pmem
+        self.server_managers[-1].manager.job.discover_nvme.value = nvme
+        self.server_managers[-1].manager.job.discover_net.value = net
+
+        self.start_server_managers()
+
+    def test_dmg_config_generate_1(self):
+        """
+        JIRA ID: DAOS-5986
+
+        Test Description:
+        Verify that dmg can generate an accurate configuration file.
```

Comment:
would be useful to have a description of what the subtests are actually doing, i.e. what parameters are going in, min_pmem etc

Reply:
Will do